ddtrace/internal/ci_visibility/_api_client.py

```diff
@@ -0,0 +1,521 @@
+    __test__ = False
+    pass
+
+
+@dataclasses.dataclass(frozen=True)
+class EarlyFlakeDetectionSettings:
+    enabled: bool = False
+    slow_test_retries_5s: int = 10
+    slow_test_retries_10s: int = 5
+    slow_test_retries_30s: int = 3
+    slow_test_retries_5m: int = 2
+    faulty_session_threshold: float = 30.0
+
+
+@dataclasses.dataclass(frozen=True)
+class TestVisibilityAPISettings:
+    __test__ = False
+    coverage_enabled: bool = False
+    skipping_enabled: bool = False
+    require_git: bool = False
+    itr_enabled: bool = False
+    flaky_test_retries_enabled: bool = False
+    early_flake_detection: EarlyFlakeDetectionSettings = dataclasses.field(default_factory=EarlyFlakeDetectionSettings)
+
+
+@dataclasses.dataclass(frozen=True)
+class ITRData:
+    correlation_id: t.Optional[str] = None
+    covered_files: t.Optional[t.Dict[str, CoverageLines]] = None
+    skippable_items: t.Set[t.Union[InternalTestId, TestSuiteId]] = dataclasses.field(default_factory=set)
+
+
+class _SkippableResponseMeta(TypedDict):
+    coverage: t.Dict[str, str]
+    correlation_id: str
+
+
+class _SkippableResponseDataItemAttributes(TypedDict):
+    name: str
+    suite: str
+    parameters: str
+    configurations: t.Dict[str, t.Any]
+
+
+class _SkippableResponseDataItem(TypedDict):
+    type: str
+    attributes: _SkippableResponseDataItemAttributes
+
+
+class _SkippableResponse(TypedDict):
+    data: t.List[_SkippableResponseDataItem]
+    meta: _SkippableResponseMeta
+
+
+def _get_test_id_from_skippable_test(
+    skippable_test: _SkippableResponseDataItem, ignore_parameters: bool
+) -> InternalTestId:
+    test_type = skippable_test["type"]
+    if test_type != TEST:
+        raise ValueError(f"Test type {test_type} is not expected test type {TEST}")
+    module_id = TestModuleId(skippable_test["attributes"]["configurations"]["test.bundle"])
+    suite_id = TestSuiteId(module_id, skippable_test["attributes"]["suite"])
+    test_name = skippable_test["attributes"]["name"]
+    test_parameters = None if ignore_parameters else skippable_test["attributes"].get("parameters")
+    return InternalTestId(suite_id, test_name, test_parameters)
+
+
+def _get_suite_id_from_skippable_suite(skippable_suite: _SkippableResponseDataItem) -> TestSuiteId:
+    suite_type = skippable_suite["type"]
+    if suite_type != SUITE:
+        raise ValueError(f"Test type {suite_type} is not expected test type {SUITE}")
+
+    module_id = TestModuleId(skippable_suite["attributes"]["configurations"]["test.bundle"])
+    return TestSuiteId(module_id, skippable_suite["attributes"]["suite"])
+
+
+def _parse_covered_files(covered_files_data: t.Dict[str, str]) -> t.Optional[t.Dict[str, CoverageLines]]:
+    covered_files = {}
+    parse_errors = 0
+    for covered_file, covered_lines_bytes in covered_files_data.items():
+        try:
+            covered_lines = CoverageLines.from_bytearray(bytearray(b64decode(covered_lines_bytes)))
```

Comment:
I was going to say that we already have a [`from_b64_string` method](https://github.com/DataDog/dd-trace-py/blob/ceea872654b11cdd2549d92fbca48561076ecfe4/ddtrace/internal/test_visibility/coverage_lines.py#L89C9-L89C24) in `CoverageLines`, but it turns out it's missing the `b64decode()` there (which I think is wrong).

Reply:
😓 that's because I started adding that method and the decided it wasn't worth adding as a new to the API.